execution/function/histogram.go

```diff
@@ -127,14 +137,16 @@ func (o *histogramOperator) processInputSeries(vectors []model.StepVector) ([]mo
 
 		step := o.pool.GetStepVector(vector.T)
 		for i, stepBuckets := range o.seriesBuckets {
-			// We need at least2 buckets to calculate a quantile.
-			if len(stepBuckets) < 2 {
+			// We only have one bucket so it needs to be NaN.
+			// Or, if we are after how many scalar points we have.
+			if len(stepBuckets) == 1 || stepIndex >= len(o.scalarPoints) {
+				step.SampleIDs = append(step.SampleIDs, uint64(i))
+				step.Samples = append(step.Samples, math.NaN())
 				continue
 			}
 
-			if stepIndex >= len(o.scalarPoints) {
-				step.SampleIDs = append(step.SampleIDs, uint64(i))
-				step.Samples = append(step.Samples, math.NaN())
+			// We need at least 2 buckets to calculate a quantile.
+			if len(stepBuckets) < 2 {
```

Comment:
Good point, `stepBuckets` will always be of length 1 or greater.

Reply:
Actually, I've hit the condition `len(stepBuckets) == 0` somehow. It leads to the following crash:

```
runtime error: index out of range [-1]
...
github.com/thanos-community/promql-engine/execution/function.bucketQuantile(0x3feccccccccccccd, {0x0?, 0x0, 0x0})\n\t/ho
me/giedrius/go/pkg/mod/github.com/vinted/promql-engine@v0.0.0-20221205101229-9ef452546ade/execution/function/quantile.go:67 +0x3c9\ngithub.com/thanos-community/promql-engine/execution/function.(*histogra
mOperator).processInputSeries(0xc079ab7550, {0xc0ca9fad80, 0xa, 0xa})\n\t/home/giedrius/go/pkg/mod/github.com/vinted/promql-engine@v0.0.0-20221205101229-9ef452546ade/execution/function/histogram.go:145 +
0x725\ngithub.com/thanos-community/promql-engine/execution/function.(*histogramOperator).Next(0xc079ab7550, {0x2ba6f60, 0xc079a9b040})\n\t/home/giedrius/go/pkg/mod/github.com/vinted/promql-engine@v0.0.0-
20221205101229-9ef452546ade/execution/function/histogram.go:114
...
```

Trying to dig how this could happen.